### PR TITLE
fix(custom-query): Introduce custom query feature(osio-1722)

### PR DESCRIPTION
**Pull request by sanbornsen**
### [ ] What does this PR do? _[Mandatory]_
_Note: If you are still working on the changes, prefix the PR title with "WIP" and add the label "DO NOT MERGE"_

Introduce Custom Query Feature
- List custom queries for a space
- Create custom query
- Clicking on custom query should update the url and filter accordingly

### [ ] What issue/task does this PR references? _[Mandatory]_

https://openshift.io/openshiftio/openshiftio/plan/detail/1722

### [ ] Are the tests Included? _[Mandatory]_

Yes (in the E2E tests PR)

### CODE REVIEW

File: src/app/components_ngrx/toolbar-panel/toolbar-panel.component.ts [hunk 1]
```diff
@@ -42,8 +42,10 @@ import { GroupTypeUI } from './../../models/group-types.model';
 // ngrx stuff
 import { Store } from '@ngrx/store';
 import { AppState } from './../../states/app.state';
+import * as CustomQueryActions from './../../actions/custom-query.actions';
 import * as FilterActions from './../../actions/filter.actions';
 import * as SpaceActions from './../../actions/space.actions';
+import { CustomQueryModel } from 'src/app/models/custom-query.model';
```
Comment (nimishamukherjee) [inline src/app/components_ngrx/toolbar-panel/toolbar-panel.component.ts]: @sanbornsen remove this. I think this got added automatically. This causes the unit test to fail.

**Comment by fabric8cd** [comment]
@sanbornsen fabric8/fabric8-ui:SNAPSHOT-PR-2512-3 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2512-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@sanbornsen fabric8/fabric8-ui:SNAPSHOT-PR-2512-5 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2512-fabric8-planner.badger.fabric8.io

**Comment by Raunak1203** [comment]
@sanbornsen clicking on custom query unchecked the 'Show Tree'. Is this intended behaviour or a bug?

**Comment by rgarg1** [comment]
@Raunak1203 @sanbornsen We had this discussion earlier and I still believe that we should save the value of 'Show Tree' as well as "Show Completed" as part of custom queries. If this id still not agreed on, we should have further discussions.

**Comment by jarifibrahim** [comment]
@sanbornsen You've committed `fabric8-test` repo. You need to remove it.

**Comment by nimishamukherjee** [comment]
@Raunak1203 please check
- Fixed long query title display on the side panel
- Save custom query on enter on the input title field

@jarifibrahim removed the fabric8-test folder

@rgarg1 please add the discussion around inclusion of showTree in today's (Apr 10th) Cabal discussion.

**Comment by Raunak1203** [comment]
@nimishamukherjee  if 2 query title have same 15characters, then how should we select the right query?

**Comment by nimishamukherjee** [comment]
@Raunak1203 can we track the below separately and merge this PR. I need to raise a PR to Sudipta each time.
> if 2 query title have same 15characters, then how should we select the right query?

**Pull request merged**